MovieBooking/Modules/Authentication/ViewController/LogInViewController.swift

```diff
@@ -0,0 +1,96 @@
+import UIKit
+import FirebaseAuth
+
+class LogInViewController: UIViewController, Storyboarded {
+    
+    // MARK: - Variables
+    var coordinator: MainCoordinator?
+    var activeTextField: UITextField? = nil
+    var authenticationViewModel = AuthenticationViewModel()
+    
+    // MARK: - Outlets
+    @IBOutlet weak var tfEmail: UITextField!
+    @IBOutlet weak var tfPassword: UITextField!
+    
+    // MARK: - UIViewController
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        bindViewModel()
+        initialData()
+    }
+    
+    // MARK: - File Private Functions
+    fileprivate func bindViewModel() {
+        authenticationViewModel.onLogInSuccess = { message in
+            print(message)
+            self.generateAlert(validation: message)
+        }
+        authenticationViewModel.logInValidationError = { message in
```

Comment:
if you are sending message in both the cases for login fail and success you can just simply take common variable like logInStatus and you can use it for both success and failure